react/components/Autocomplete/index.tsx

```diff
@@ -265,6 +327,23 @@ class AutoComplete extends React.Component<
       simulationBehavior,
     );
 
+    if (!queryFromHover) {
+      const { count, operator, misspelled } = result.data.productSuggestions;
+
+      console.log(result.data.productSuggestions);
+
+      push({
+        event: EVENT_NAME,
+        eventType: count > 0 ? EventType.Search : EventType.EmptySearch,
+        search: {
+          operator,
+          misspelled,
+          text: decodeURI(term),
+          match: count,
+        },
+      });
+    }
```

Comment:
What if we move this, and the `handle*Click` functions to a separate file? Will it be weird?
As this one file has 500+ lines, maybe dividing it into more specific files would help a bit with readability.

If you opt to go this way the functions would probably look like:

```typescript
export function handleProductClick(push: (data: PixelData) => void, page: string, productId: string, position: number): void
// or maybe
export function buildProductClickHandler(push: (data: PixelData) => void, page: string): (string, number) => void
```

It's really subjective, as we would be removing it from the component. A different approach would be a Provider with this functions:

```jsx
<AutocompleteEventProvider handleProductClick={handleProductClick} handleItemClick={handleItemClick} /* ... */ />
```

Reply:
You're right. This file is huge.

I'm keeping the handle functions on the `utils/pixel.ts` file now. Is it better?